### PR TITLE
feat(behavior_velocity_intersection_module): consider NPC vehicles driving on the wrong direction lane on intersection

### PR DESCRIPTION
**Pull request by beyzanurkaya**
## Description

fixes: https://github.com/autowarefoundation/autoware.universe/issues/3969

## Tests performed

Before this PR:
https://www.youtube.com/watch?v=85KuJ6GFbMk

After this PR:
https://www.youtube.com/watch?v=Qt-jrlWgqYQ&ab_channel=BeyzaNurKaya

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by soblin** [comment]
@beyzanurkaya Please provide the parameter for `common.consider_wrong_direction_vehicle` as well. Does this default to `false` ?

**Comment by beyzanurkaya** [comment]
@soblin I added the missing parameter. This is a new feature and this situation is not very common so I think the default value should be `false`

**Comment by takayuki5168** [comment]
@soblin Could you review the PR.

**Comment by soblin** [comment]
@beyzanurkaya Can you also create a PR in autoware_launch for the change in yaml file ?

**Comment by beyzanurkaya** [comment]
> @beyzanurkaya Can you also create a PR in autoware_launch for the change in yaml file ?

I already created a PR for this: 

[394](https://github.com/autowarefoundation/autoware_launch/pull/394)

**Comment by beyzanurkaya** [comment]
@soblin I reverted the changes you want

File: planning/behavior_velocity_intersection_module/config/intersection.param.yaml [hunk 1]
```diff
@@ -4,14 +4,14 @@
       common:
         attention_area_margin: 0.5 # [m]
         attention_area_length: 200.0 # [m]
-        attention_area_angle_threshold: 0.785 # [rad]
+        attention_area_angle_threshold: 1.5 # [rad]
```
[the remark this answers is not in the document]
Comment (beyzanurkaya) [reply planning/behavior_velocity_intersection_module/config/intersection.param.yaml]: @soblin I reverted the changes you want

**Review by soblin — APPROVED** [review]

**Comment by soblin** [comment]
@beyzanurkaya Thanks !

**Pull request merged**